api/apiroutes.go

```diff
@@ -1074,6 +1075,30 @@ func (c *appContext) getTicketPoolByDate(w http.ResponseWriter, r *http.Request)
 	writeJSON(w, tpResponse, c.getIndentQuery(r))
 }
 
+func (c *appContext) getProposalChartData(w http.ResponseWriter, r *http.Request) {
+	if c.LiteMode {
+		// not available in lite mode
+		http.Error(w, "not available in lite mode", 422)
+		return
+	}
+
+	token := m.GetProposalTokenCtx(r)
+	votesData, err := c.AuxDataSource.ProposalVotes(token)
```

Comment:
Eventually, it'd be good to cache this data and trigger a refresh on the hourly updates. Doesn't need to be done here necessarily.

Reply:
I am not sure if this is necessary. The charts data retrieved has a max of 168 records (Politieia voting happens within a week and votes data is pushed to github hourly). IMO a cache will be an overkill. It might become too expensive to maintain.